### DISCUSSION TITLE
Respect the allowedPlacements order if the floating UI doesn't overflow.

### DISCUSSION THREAD

**Discussion by psuzn** [floating-ui/floating-ui#2900] [Ideas]
I have this setup. 

```ts

  const { refs, floatingStyles, context } = useFloating({
      open: isOpen,
      onOpenChange: setIsOpen,
      middleware: [
          offset(() => {
              return {
                  mainAxis: 12,
              };
          }),
          autoPlacement({
              allowedPlacements: ["top", "right"],
          }),
      ],
  });

```

https://github.com/floating-ui/floating-ui/assets/19205305/c194152c-27ad-47b1-a0aa-4c451dbe9c97

I'd like it to place floating content on the top and then only fall back to the right position if it doesn't fit on top. 

Not sure if there is already a way to achieve this. If there is not this can be achieved by respecting the order in which the allowedPlacements are provided and only placing the item to the next position if it overflows on current.

**Comment by atomiks** [comment]
> I'd like it to place floating content on the top and then only fall back to the right position if it doesn't fit on top.

Instead of `autoPlacement`, use `flip` with `fallbackPlacements`.

**Reply by psuzn** [reply]
that works, thank you.
